core/src/main/java/org/mule/api/MuleEvent.java

```diff
@@ -322,6 +322,10 @@
 
     void clearFlowVariables();
 
+    /**
+     * @deprecated use {@code getSession()} to manipulate session state
+     */
+    @Deprecated
```

Comment:
shouldn't we wait to get sign off on the message changes spec before doing this?